Test/runtime/file-io/AbstractFileIO.dfy

```diff
@@ -0,0 +1,27 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+include "../../libraries/src/Wrappers.dfy"
+
+abstract module AbstractFileIO {
+  import opened Wrappers
+
+  export provides ReadBytesFromFile, WriteBytesToFile, Wrappers
+
+  method ReadBytesFromFile(path: string) returns (res: Result<seq<bv8>, string>) {
```

Comment:
Perhaps an additional argument is needed -- along with creating intermediate folders, should the method overwrite an existing file?

Reply:
Yeah, I'm inclined not to add any arguments yet, but to make `WriteBytesToFile` create directories as needed and overwrite existing files. I'm imagining a common use case will be to cache data in a structured way to a hierarchy of directories under a given root folder.

Reply:
> You've got excellent documentation on the methods in the runtimes, but these are the actual user-facing methods so we should have it here too!

Ack, I've added a TODO to comment the user-facing APIs on the new PR.

> In particular, will WriteBytesToFile create any intermediate directories automatically? For this extremely simple interface I expect that would be almost a hard requirement, since otherwise you have to drop down to native code to create them.
>
> Perhaps an additional argument is needed -- along with creating intermediate folders, should the method overwrite an existing file?

As implemented now, `WriteBytesToFile` always overwrites an existing file (assuming it can write to the file at all). It seems like a good idea to me that `WriteBytesToFile` should create directories as needed; I'll tackle this in the new PR as well.